src/queue/QueueNode.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * The queue package contains our Queue, QueueTest, QueueNode and QueueTest 
+ * classes.
+ * 
+ * @author Scott McDonald and Jon Cronin
+ */
+package queue;
+
+/**
+ * The QueueNode class is used to represent each element in a Queue.
+ * 
+ * @author Scott McDonald and Jon Cronin
+ */
+public class QueueNode<E> {
```

Comment:
This is the java generic type notation. You can have as many generic types in a class as you need. Users can then pass in any object for that type. It is similar to what you did using "object" in linkedList, but different because you can do some type checking on the generic types at runtime. I think it is the preferred style for Java API classes. The collections classes like ArrayList all use it. Core Java for the Impatient covers them well in chapter 6 "Generic Programming".